core/src/main/scala/chisel3/probe/package.scala

```diff
@@ -68,10 +68,20 @@ package object probe extends SourceInfoDoc {
     }
   }
 
+  /** Pad Data if it supports padding */
+  private def padData[T <: Data](data: T, width: Int)(implicit sourceInfo: SourceInfo): T = {
+    if (!data.width.known) Builder.error("Data width unknown")
+    if (data.width.get > width) Builder.error(s"Data width ${data.width.get} is larger than $width.")
```

Comment:
```suggestion
    data.widthOption match {
      case None => Builder.error("Data width unknown")
      case Some(w) =>
        if (w > width) Builder.error(s"Data width $w is larger than $width.")
    }
```
As written, an unknown width will still result in throwing an exception on that `.get`. That wouldn't be the worst thing in the world but it will immediately kill elaboration whereas these are recoverable errors so it is better to continue elaboration so that we can aggregate any warnings or errors from later elaboration.